integrations/gtag/lib/index.js

```diff
@@ -0,0 +1,893 @@
+  });
+};
+
+/**
+ * User Logged In - Enhanced Ecommerce
+ *
+ * @param {Track} track
+ */
+
+GTAG.prototype.loggedInEnhanced = function(track) {
+  var props = track.properties();
+  push('event', 'login', { method: props.method });
+};
+
+/**
+ * User signed Up - Enhanced Ecommerce
+ *
+ * @param {Track} track
+ */
+
+GTAG.prototype.signedUpEnhanced = function(track) {
+  var props = track.properties();
+  push('event', 'sign_up', { method: props.method });
+};
+
+/**
+ * Lead Generated - Enhanced Ecommerce
+ *
+ * @param {Track} track
+ */
+
+GTAG.prototype.leadGeneratedEnhanced = function(track) {
+  push('event', 'generate_lead', {
+    transaction_id: track.id(),
+    value: track.price(),
+    currency: track.currency()
+  });
+};
+
+/**
+ * Exception - Enhanced Ecommerce
+ *
+ * @param {Track} track
+ */
+
+GTAG.prototype.exceptionOccuredEnhanced = function(track) {
+  var props = track.properties();
+  push('event', 'exception', {
+    description: props.description,
+    fatal: props.fatal
+  });
+};
+
+/**
+ * Timing Completed - Enhanced Ecommerce
+ *
+ * @param {Track} track
+ */
+
+GTAG.prototype.timingCompletedEnhanced = function(track) {
+  push('event', 'timing_complete', {
+    name: track.name(),
+    value: track.value()
+  });
+};
+
+/**
+ * Track page view event.
+ *
+ * @api private
+ * @param { Page } page
+ * @param  opt
+ * @param  props
+ */
+
+function trackPageViewEvent(page, opts, props) {
+  var nonInteraction =
+    props && props.nonInteraction !== undefined
+      ? !!props.nonInteraction
+      : !!(opts && opts.nonInteraction);
+  push('event', 'page_view', {
+    page_title: page.title,
+    page_location: page.url,
+    page_path: page.path,
+    non_interaction: nonInteraction
+  });
+}
+
+/**
+ * Set custom dimensions and metrics.
+ *
+ * @api private
+ * @param options
+ */
+
+function setCustomDimensionsAndMetrics(options) {
+  var gaOptions = options.gaOptions || {};
+  if (gaOptions && Object.keys(gaOptions).length) {
+    if (gaOptions.setAllMappedProps) {
+      // set custom dimension and metrics if present
+      // REF: https://developers.google.com/analytics/devguides/collection/gtagjs/custom-dims-mets
+
+      var customMap = merge(gaOptions.dimensions, gaOptions.metrics);
+      if (options.gaWebMeasurementId) {
+        push('config', options.gaWebMeasurementId, {
+          custom_map: customMap
+        });
+      }
+      if (options.gaWebAppMeasurementId) {
+        push('config', options.gaWebAppMeasurementId, {
+          custom_map: customMap
+        });
+      }
+    }
+  }
+}
+
+/**
+ * Enhanced ecommerce format data for checkout.
+ *
+ * @api private
+ * @param {Track} track
+ */
+function extractCheckoutOptions(track) {
+  var props = track.properties();
+  var total = track.total() || track.revenue() || 0;
+  var coupon = track.proxy('properties.coupon');
+  var options = reject([
+    track.proxy('properties.paymentMethod'),
+    track.proxy('properties.shippingMethod')
```

Comment:
I think here if you just did `track.paymentMethod()` or `track.shippingMethod()` this will return either 
`this.proxy('properties.payment_method') || this.proxy('properties.paymentMethod');` or 
` return this.proxy('properties.shipping_method') || this.proxy('properties.shippingMethod');` respectively. That way we can check for camel and snake case. These are two methods we have on in our track facade library.

Reply:
Done, It's working fine using facade methods but earlier we was matching the parity with the existing GA integration.